kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/ExpressionUtils.java

```diff
@@ -64,4 +65,22 @@ public static Expression getUnaryChild(Expression expression) {
             format("%s: expected one inputs, but got %s", expression, children.size()));
         return children.get(0);
     }
+
+    /*
+     * Utility method to combine the given predicates with AND
+     */
+    public static Predicate combineWithAndOp(Predicate left, Predicate right) {
+        String leftName = left.getName().toUpperCase();
+        String rightName = right.getName().toUpperCase();
+        if (leftName.equals("ALWAYS_FALSE") || rightName.equals("ALWAYS_FALSE")) {
+            return ALWAYS_FALSE;
+        }
+        if (leftName.equals("ALWAYS_TRUE")) {
+            return right;
+        }
+        if (rightName.equals("ALWAYS_TRUE")) {
+            return left;
+        }
+        return new And(left, right);
+    }
```

Comment:
Something feels off here. Why are we doing string comparisons? Isn't everything strongly typed?

Reply:
Our expressions name (`=`, `like` etc) based , we don't enforce any restrictions on the expression name to allow the connector to pass through Kernel any arbitrary expressions which its own expression handler can support